duckduckgo/iOS
# Favorites overlay

Task/Issue URL: https://app.asana.com/0/715106103902962/1126103212632242

**Description**:
Adds overlay above web view when search bar is focused and home screen works in centered search with favorites mode.

Extras:
* Added highlight for Favorites/Add Fav cells.
* Long press on Add Fav cell will invoke add favorite dialog box. 

**Steps to test this PR**:
Things to check:
1. Home screen favorites view should work without changes. 
2. Overlay should work only in "centered search with favorites mode" and there are some favorites set.
3. Overlay should not be editable.
4. Autocomplete should work as expected.
5. Canceling search bar interaction should dismiss overlay.
6. Rotating should dismiss overlay.
7. On iPad grid width should match what is seen on the home screen.
8. In case there are many items in favorites, overlay should allow scrolling.

Additionally:
1. Open app, enter omni bar so the favorites are displayed (you can also type something so autocomplete is presented).
2. Leave the app.
3. Try to enter the app from DDG extensions (bookmarks, search, clear data).


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
